### PR TITLE
feat: move js code implemented directly on library.html (dynamo side) to librariejs

### PR DESCRIPTION
**Pull request by Enzo707**
We have implemented javascript code, at some point in the past, related to library web component functionalities, directly into Dynamo. So now that we want to consume library as npm package, we need to move that functionalities keeping all library's responsibilities together one side in order to just install the package into Dynamo side.

### CODE REVIEW

**Comment by RobertGlobant20** [comment]
@Enzo707  with this changes, could you add a GIF showing the expected behavior when running the guided tours in Dynamo?
image

**Comment by Enzo707** [comment]
> @Enzo707 with this changes, could you add a GIF showing the expected behavior when running the guided tours in Dynamo? image

@RobertGlobant20  I've tested and the tour guide works just fine

**Comment by mjkkirschner** [comment]
looks like sonar cloud has found a few issues.

**Comment by Enzo707** [comment]
> looks like sonar cloud has found a few issues.

@mjkkirschner Yes, we didn't implement sonarcloud validations from the start so every time we change a file the scan throws a bunch of issues. The one I couldn't address is related to a function we expose to be used on Dynamo side so seems to be defined but not used. Can we just merge it anyway or how should we proceed?

**Comment by QilongTang** [comment]
Hi @Enzo707 Thanks for proposing this PR, is there a pairing PR on Dynamo side to remove these code?

File: index.html [hunk 1]
```diff
@@ -5,75 +5,458 @@
...
+                    window.chrome.webview.postMessage(JSON.stringify({"func":"logEventsToInstrumentation","data":["Filter-Categories",categories.join(",")]}));
             });
 
-            libController.on(libController.SectionIconClickedEventName, function (sectionText) {
-                console.log(sectionText, "icon clicked");
-                return null;
-            });
-            libController.on(libController.FilterCategoryEventName, function (item) {
-                console.log(item);
+            //This will call the NextStep() function located in the LibraryViewController
```
Comment (QilongTang) [inline index.html:251]: This is called by Dynamo code right?

File: index.html [hunk 2]
```diff
@@ -5,75 +5,458 @@
...
+</style>
+</head>
+
+<body onresize="bodyResizeEvent()">
+    <div id="overlay"></div>
+    <!-- Placeholders must exist before all other scripts that try to access them -->
+    <div class="OuterMostContainer" id="libraryContainerPlaceholder"></div>
+    <!-- The main library view component -->
+    <script>
+        LIBPLACEHOLDER
```
[the remark this answers is not in the document]
Comment (Enzo707) [reply index.html:111]: @QilongTang My understanding is that this is used on Dynamo side to render library there after icons urls update.
If it´s ok, I think we can leave it like this for now and then once we have all working as an npm package we can go forward doing major changes according to @mjkkirschner proposal for web components.

File: index.html (hunk 1 above)
Comment (Enzo707) [reply index.html:251]: > This is called by Dynamo code right?

Yes, it is.

File: index.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (QilongTang) [reply index.html:111]: Thanks for confirming @Enzo707 This is fine for me

**Comment by Enzo707** [comment]
> Hi @Enzo707 Thanks for proposing this PR, is there a pairing PR on Dynamo side to remove these code?

Not yet, I wanted to bring this up first. If you agree I can created a PR on Dynamo repo removing this code.

**Comment by QilongTang** [comment]
@Enzo707 Also notice you may want to bump up the version number for this change so we can re-publish

**Comment by Enzo707** [comment]
> @Enzo707 Also notice you may want to bump up the version number for this change so we can re-publish

Sure, I can do that. I have to rebase this branch also, as we already merged Roberto's PR related to library search.

**Comment by QilongTang** [comment]
> > @Enzo707 Also notice you may want to bump up the version number for this change so we can re-publish
> 
> Sure, I can do that. I have to rebase this branch also, as we already merged Roberto's PR related to library search.

Thanks

**Pull request merged**